amp_conf/htdocs/admin/libraries/BMO/ConfigFile.class.php

```diff
@@ -83,25 +83,21 @@ public function removeEntry($section, $key, $val = null) {
 		if (!isset($this->config->ProcessedConfig[$section]))
 			throw new \Exception("Tried to remove key $key from section $section, but that section doesn't exist");
 
-		if (is_array($this->config->ProcessedConfig[$section][$key])) {
-			if ($val == null)
-				throw new \Exception("Sorry, you can't delete an entire section this way, as it's likely a bug");
-
-			// Ok, it's buried in here somewhere. Sigh.
-			foreach ($this->config->ProcessedConfig[$section][$key] as $id => $v) {
-				if ($v == $val) {
-					// Yay, found it.
-					unset($this->config->ProcessedConfig[$section][$key][$id]);
-					// Note, keep going - could be dupes. Don't break here.
-				}
-			}
-
-			// Have we deleted everything from that $key?
-			if (count($this->config->ProcessedConfig[$section][$key]) == 0)
+		if (isset($this->config->ProcessedConfig[$section][$key])) {
+			if (is_array($this->config->ProcessedConfig[$section][$key])) {
```

Comment:
I didn't combine the checks because there's still an `else` statement below at line 97. If I combined it, I would have had to do an `elseif isset()` instead. Just seemed easier to wrap it all and indent a level.